haystack/pipelines/base.py

```diff
@@ -570,6 +403,7 @@ def get_node(self, name: str) -> Optional[BaseComponent]:
         component = graph_node["component"] if graph_node else None
         return component
 
+    # FIXME unused and untested. In which cases do we need to set nodes? Can this be removed?
```

Comment:
Good question. As it doesn't set the name of the component, a call of this method would probably break `Pipeline.get_config()`. We should at least set the name if we want to keep it.